src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala

```diff
@@ -24,156 +27,159 @@ case class CromwellRuntimeAttributes(docker: Option[String],
-    val FailOnStderr = false
-    val ContinueOnReturnCode = ContinueOnReturnCodeSet(Set(0))
-    val Memory = 2.0
-    val Preemptible = false
-    val Zones = Vector("us-central1-a")
-  }
-
-  private def validateRuntimeAttributes(attributeMap: AttributeMap): ValidationNel[String, CromwellRuntimeAttributes] = {
-    val docker = attributeMap.get(DOCKER)
-    val zones = attributeMap.get(DEFAULT_ZONES) map { _.split("\\s+").toVector } getOrElse Defaults.Zones
+  val LocalizationDisk = LocalDisk(LocalDiskName, DiskType.SSD, sizeGb=10)
+
+  val defaults: CromwellRuntimeAttributes = {
+    // .get below because we assume that the default values are valid
+    // There's a unit test in CromwellRuntimeAttributeSpec to test these defaults
+    validateRuntimeAttributes(Map(
+      "cpu" -> WdlInteger(1),
+      "defaultDisks" -> WdlString(LocalizationDisk.toString),
+      "defaultZones" -> WdlString("us-central1-a"),
+      "continueOnReturnCode" -> WdlInteger(0),
+      "failOnStderr" -> WdlBoolean.False,
+      "preemptible" -> WdlBoolean.False,
+      "memory" -> WdlString("2GB")
+    )).get
+  }
+
+  private def validateRuntimeAttributes(attributes: Map[String, WdlValue]): Try[CromwellRuntimeAttributes] = {
+    val attributeMap = attributes.mapKeys(RuntimeKey.from)
+    val docker = validateDocker(attributeMap.get(DOCKER))
+    val zones = validateZone(attributeMap.get(DEFAULT_ZONES))
     val failOnStderr = validateFailOnStderr(attributeMap.get(FAIL_ON_STDERR))
-    val continueOnReturnCode = validateContinueOnReturnCode(attributeMap.getSeq(CONTINUE_ON_RETURN_CODE))
+    val continueOnReturnCode = validateContinueOnReturnCode(attributeMap.get(CONTINUE_ON_RETURN_CODE))
     val cpu = validateCpu(attributeMap.get(CPU))
     val preemptible = validatePreemptible(attributeMap.get(PREEMPTIBLE))
     val disks = validateLocalDisks(attributeMap.get(DEFAULT_DISKS))
-    val memory = validateMemory(attributeMap.get(MEMORY))
+    val memory = RuntimeAttributes.validateMemoryValue(attributeMap.getOrElse(MEMORY, WdlString(s"${defaults.memoryGB} GB"))) match {
+      case scala.util.Success(x) => x.to(MemoryUnit.GB).amount.successNel
+      case scala.util.Failure(x) => x.getMessage.failureNel
+    }
 
-    (failOnStderr |@| continueOnReturnCode |@| cpu |@| preemptible |@| disks |@| memory) {
-      CromwellRuntimeAttributes(docker, zones, _, _, _, _, _, _)
+    (docker |@| zones |@| failOnStderr |@| continueOnReturnCode |@| cpu |@| preemptible |@| disks |@| memory) {
+      new CromwellRuntimeAttributes(_, _, _, _, _, _, _, _)
+    } match {
+      case Success(x) => scala.util.Success(x)
+      case Failure(nel) => scala.util.Failure(new IllegalArgumentException(nel.list.mkString("\n")))
     }
   }
 
-  private def validateCpu(value: Option[String]): ValidationNel[String, Long] = {
-    value map validateLong getOrElse Defaults.Cpu.successNel
+  private def validateCpu(cpu: Option[WdlValue]): ValidationNel[String, Int] = {
```

Comment:
I recognize it's preexisting but could you change ValidationNel to ErrorOr (throughout the PR)